src/server/_common.py

```diff
@@ -36,6 +63,24 @@ def connect_db():
         app.logger.error('database connection error', exc_info=True)
         raise DatabaseErrorException()
 
+@app.before_request
+def log_request_info():
+    app.logger.info('Received API request: %s', str(dict(method=request.method, path=request.full_path, args=request.args)))
+
+@app.before_request
+def time_request_start():
+    t = time.time()
+    g._request_start_time = t
+
+@app.after_request
+def time_request_end(response):
+    t = time.time()
+    total_time = t - g._request_start_time
+    # Convert to milliseconds
+    total_time = total_time * 1000
+    app.logger.info('API request elapsed time: %s', str(dict(method=request.method, path=request.full_path, args=request.args, time=total_time)))
```

Comment:
- same comment as above about units and value formatting
- should also add info from the response object like length and http code